src/main/java/bookstore/service/impl/CartItemServiceImpl.java

```diff
@@ -0,0 +1,52 @@
+package bookstore.service.impl;
+
+import bookstore.dto.cart.CartItemCreateDto;
+import bookstore.entity.book.Book;
+import bookstore.entity.cart.Cart;
+import bookstore.entity.cart.CartItem;
+import bookstore.mapper.CartItemMapper;
+import bookstore.repository.BookRepository;
+import bookstore.repository.CartItemRepository;
+import bookstore.service.CartItemService;
+import jakarta.persistence.EntityNotFoundException;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@Service
+@RequiredArgsConstructor
+public class CartItemServiceImpl implements CartItemService {
+    private final BookRepository bookRepository;
+    private final CartItemMapper cartItemMapper;
+    private final CartItemRepository cartItemRepository;
+
+    @Override
+    @Transactional
+    public void saveCartItem(CartItemCreateDto addToCartDto, Cart cart) {
+        Book book = bookRepository.findById(addToCartDto.book_id()).orElseThrow(
+                () -> new EntityNotFoundException("Book was not found with id: "
+                        + addToCartDto.book_id())
+        );
+
+        CartItem cartItem = cartItemMapper.toModel(cart, book, addToCartDto.quantity());
+        cartItemRepository.saveAndFlush(cartItem);
+    }
```

Comment:
Seems like in "toModel(cart, book, addToCartDto.quantity())" you add new cart item every time endpoint is used. It may lead to situation where you will have multiple cart items with exactly the same book in the same shopping cart. It may be better to first check if there already is a cartitems for this book and update its quantity instead. Although it's for you to decide